### PR TITLE
[Merged by Bors] - feat(GroupTheory): some lemmas about blocks

### PR DESCRIPTION
**Pull request by YaelDillies**
From Kneser (LeanCamCombi)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/GroupAction/Blocks.lean [hunk 1]
```diff
@@ -149,6 +170,19 @@ lemma IsFixedBlock.isInvariantBlock (hB : IsFixedBlock G B) : IsInvariantBlock G
 
 end SMul
 
+section Monoid
```
Comment (AntoineChambert-Loir) [inline Mathlib/GroupTheory/GroupAction/Blocks.lean:173]: can you add the additive version?

File: Mathlib/GroupTheory/GroupAction/Blocks.lean
Comment (AntoineChambert-Loir) [inline Mathlib/GroupTheory/GroupAction/Blocks.lean:1]: LGTM.
For my own information, can you tell me in what context one is led to compare a block with a set-translate of that block?

Comment (YaelDillies) [reply Mathlib/GroupTheory/GroupAction/Blocks.lean:1]: Do you specifically mean `IsBlock.disjoint_smul_right`? If so, I originally thought I needed that lemma, but turns out I did not.

If instead you mean all lemmas in this PR more generally, the use case comes from the proof of Kneser's addition theorem, where we have two sets `s` and `t` and look at cosets of `Stab (s * t)`. The proof is a long series of manipulations of those cosets

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**